ThePensionsRegulator.Frontend.Umbraco.Tests/Services/TprDividerViewInterceptorTests.cs

```diff
@@ -0,0 +1,220 @@
+
+			// Assert
+			Assert.That(blockViewModel.OpenGridRowAndColumn, Is.EqualTo(contentAlias == GovUkElementTypeAliases.PageHeading || hasLegendAsHeading || hasLabelAsHeading));
+		}
+
+		[TestCase(GovUkElementTypeAliases.PageHeading, GovUkElementTypeAliases.PageHeadingSettings, false, false)]
+		[TestCase(GovUkElementTypeAliases.Checkboxes, GovUkElementTypeAliases.CheckboxesSettings, true, false)]
+		[TestCase(GovUkElementTypeAliases.Checkboxes, GovUkElementTypeAliases.CheckboxesSettings, false, false)]
+		[TestCase(GovUkElementTypeAliases.DateInput, GovUkElementTypeAliases.DateInputSettings, true, false)]
+		[TestCase(GovUkElementTypeAliases.DateInput, GovUkElementTypeAliases.DateInputSettings, false, false)]
+		[TestCase(GovUkElementTypeAliases.Fieldset, GovUkElementTypeAliases.FieldsetSettings, true, false)]
+		[TestCase(GovUkElementTypeAliases.Fieldset, GovUkElementTypeAliases.FieldsetSettings, false, false)]
+		[TestCase(GovUkElementTypeAliases.Radios, GovUkElementTypeAliases.RadiosSettings, true, false)]
+		[TestCase(GovUkElementTypeAliases.Radios, GovUkElementTypeAliases.RadiosSettings, false, false)]
+		[TestCase(GovUkElementTypeAliases.FileUpload, GovUkElementTypeAliases.FileUploadSettings, false, true)]
+		[TestCase(GovUkElementTypeAliases.FileUpload, GovUkElementTypeAliases.FileUploadSettings, false, false)]
+		[TestCase(GovUkElementTypeAliases.Select, GovUkElementTypeAliases.SelectSettings, false, true)]
+		[TestCase(GovUkElementTypeAliases.Select, GovUkElementTypeAliases.SelectSettings, false, false)]
+		[TestCase(GovUkElementTypeAliases.Textarea, GovUkElementTypeAliases.TextareaSettings, false, true)]
+		[TestCase(GovUkElementTypeAliases.Textarea, GovUkElementTypeAliases.TextareaSettings, false, false)]
+		[TestCase(GovUkElementTypeAliases.TextInput, GovUkElementTypeAliases.TextInputSettings, false, true)]
+		[TestCase(GovUkElementTypeAliases.TextInput, GovUkElementTypeAliases.TextInputSettings, false, false)]
+		public void Block_preceding_component_where_class_is_added_sets_CloseGridRowAndColumn_true(string contentAlias, string settingsAlias, bool hasLegendAsHeading, bool hasLabelAsHeading)
+		{
+			// Arrange
+			var isPageHeadingProperty = hasLegendAsHeading ? GovUkPropertyAliases.FieldsetLegendIsPageHeading : hasLabelAsHeading ? GovUkPropertyAliases.LabelIsPageHeading : null;
+
+			var blockViewModel = CreateBlockView("content", "settings", false);
+			blockViewModel.NextBlock = CreateBlock(contentAlias, settingsAlias);
+			if (isPageHeadingProperty is not null)
+			{
+				Mock.Get(blockViewModel.NextBlock.Settings).SetupUmbracoBooleanPropertyValue(isPageHeadingProperty, true);
+			}
+
+			var interceptor = new TprDividerViewInterceptor();
+
+			// Act
+			interceptor.InterceptBlockView(blockViewModel);
+
+			// Assert
+			Assert.That(blockViewModel.CloseGridRowAndColumn, Is.EqualTo(contentAlias == GovUkElementTypeAliases.PageHeading || hasLegendAsHeading || hasLabelAsHeading));
```

Comment:
It would be better if the Asserts did not contain logic e.g. contentAlias == GovUkElementTypeAliases.PageHeading || hasLegendAsHeading || hasLabelAsHeading , but rather a pre-calculated expected value fed in as a parameter to the test. This will help a) write the test cases like a blackbox, separating your knowledge of the code from the business requirements b) encourage you to write more negative tests c) can help test and find bugs in the unit test itself.